Table reader

Uggah opened the pull request.

kknoepfle committed:
add test dbs

kknoepfle committed:
basic data classes
Column
Constraint
ConstraintType
Table

kknoepfle committed:
basic data classes
Column
Constraint
ConstraintType
Table

kknoepfle committed:
TableReader

kknoepfle committed:
TableReader

kknoepfle committed:
basic MigrationManager

kknoepfle committed:
basic MigrationManager

kknoepfle committed:
work around broken validation

add migration manager

kknoepfle committed:
work around broken validation

add migration manager

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/databasemodel/Table.java:
```diff
@@ -0,0 +1,21 @@
+public class Table {
+
+  final private String name;
+  private ConnectionManager connectionManager;
```
Why does a table need a connection manager?

Joystick01 replied:
is removed

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/databasemodel/Table.java:
```diff
@@ -0,0 +1,21 @@
+  final private String name;
+  private ConnectionManager connectionManager;
+  private List<Column> columns = new ArrayList<>();
+  private Constraint[] constraints;
```
Why not use a List?

Joystick01 replied:
i thought an array might be best practice due to memory efficency

is changed

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/databasemodel/Table.java:
```diff
@@ -0,0 +1,21 @@
+
+  final private String name;
+  private ConnectionManager connectionManager;
+  private List<Column> columns = new ArrayList<>();
```
Can't this be final?

Joystick01 replied:
is changed

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/MigrationManager.java:
```diff
@@ -0,0 +1,35 @@
+@NoArgsConstructor
+public final class MigrationManager {
+
+  private static MigrationManager INSTANCE;
```
Final, please.

Joystick01 replied:
not possible

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReader.java:
```diff
@@ -0,0 +1,19 @@
+
+  ;
+
+  public abstract Table[] readTables() throws SQLException;
```
Why not return a List<Table>, Set<Table> or Collection<Table>?

Joystick01 replied:
i thought an array might be best practice due to memory efficency

is changed

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReaderFactory.java:
```diff
@@ -0,0 +1,39 @@
+@NoArgsConstructor
+public final class TableReaderFactory {
+
+  private static TableReaderFactory INSTANCE;
```
Final, please.

Joystick01 replied:
not possible

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReaderFactory.java:
```diff
@@ -0,0 +1,39 @@
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
```
`@NoArgsConstructor(accessLevel = AccessLevel.PRIVATE)`

Joystick01 replied:
is changed

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReaderFactory.java:
```diff
@@ -0,0 +1,39 @@
+      case "SQLite":
+        return new SQLiteTableReader(connectionManager);
+      default:
+        return null;
```
Shouldn't we throw an Exception here?

Joystick01 replied:
is changed

Uggah commented on src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReaderFactory.java:
```diff
@@ -0,0 +1,39 @@
+
+  public TableReader getTableReader() throws SQLException {
+
+    System.out.println(
```
Please use logging...

Joystick01 replied:
is removed, was a debugging artefact

kknoepfle committed:
apply review suggestions

kknoepfle committed:
apply review suggestions

Uggah committed:
Enforce naming conventions

Uggah committed:
Improve code quality

Uggah committed:
Move testdata

Uggah committed:
Merge remote-tracking branch 'origin/TableReader' into TableReader

# Conflicts:
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/DBWarpCLI.java
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/MigrationManager.java
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/SQLiteTableReader.java
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReader.java
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/TableReaderFactory.java

Uggah committed:
Remove duplicates

Uggah committed:
Enforce tracing

kknoepfle committed:
turn MigrationManager into normal class

Uggah committed:
Add functionality, tests and improve architecture

- Tables can now have a schema
- Retrieving tables and columns can now be delegated using the template method pattern if necessary

Uggah committed:
Add missing test database

Uggah committed:
Draft PostgreSQLTableReader implementation

kknoepfle committed:
remove non constraints

kknoepfle committed:
formatting

kknoepfle committed:
add getColumnByName

kknoepfle committed:
retrive unique constraints

kknoepfle committed:
formatting

kknoepfle committed:
extend table reader by retrieve constraints

kknoepfle committed:
update constraint model

Uggah committed:
Draft Constraint reading

kknoepfle committed:
Merge remote-tracking branch 'origin/TableReader' into TableReader

# Conflicts:
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/migration/tablereader/DefaultTableReader.java
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/model/constraints/Constraint.java
#	src/main/java/de/hdm_stuttgart/mi/dbad/dbwarp/model/constraints/ConstraintType.java

kknoepfle committed:
adjust UniqueConstraint retrieval to new api

kknoepfle committed:
merge TableDescriptor and Table

kknoepfle committed:
extract table reader into separate class

kknoepfle committed:
extract ColumnReader into separate class

Uggah committed:
Clean code

- Replace statements with prepared statements
- Close prepared statments in close method
- Add table to Column and Constraint

Uggah committed:
Improve testing

- Add providers for Postgres and SQLite
- Add Test for SQLiteConstraintReader

Uggah committed:
Use SQLiteProvider for SQLiteTableReaderIntegrationTest

Uggah committed:
Replace spring-jdbc with ScriptUtils from testcontainers

Uggah committed:
Remove unused testdata
